Assets/LeapMotionModules/UI/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,293 @@
+    /// method is called. Using more than one InteractionManager is valid, but be
+    /// sure to assign the InteractionBehaviour's desired manager appropriately.
+    /// </summary>
+    /// 
+    /// <remarks> By default, this static property contains the first InteractionManager
+    /// that has had its Awake() method called in the current scene. If an
+    /// InteractionBehaviourBase does not have a non-null interactionManager by the
+    /// time it has Start() called, it will default to using the InteractionManager
+    /// referenced here.
+    /// 
+    /// If you have multiple InteractionManagers in your scene, you should be sure to
+    /// assign InteractionBehaviours' managers appropriately. If you instantiate an
+    /// InteractionBehaviour at runtime, you should assign its InteractionManager
+    /// right after you instantiate it. </remarks>
+    public static InteractionManager singleton { get; set; }
+
+    void Awake() {
+      if (InteractionManager.singleton == null) singleton = this;
+
+      Provider = Hands.Provider;
+
+      _interactionHands[0] = new InteractionHand(this, () => { return Hands.FixedLeft; }, Chirality.Left, WorldHoverActivationRadius, WorldTouchActivationRadius);
+      _interactionHands[1] = new InteractionHand(this, () => { return Hands.FixedRight; }, Chirality.Right, WorldHoverActivationRadius, WorldTouchActivationRadius);
+
+      if (_autoGenerateLayers) {
+        AutoGenerateLayers();
+        AutoSetupCollisionLayers();
+      }
+    }
+
+    void Start() {
+      if (Provider == null) {
+        Debug.LogError("[InteractionManager] No LeapServiceProvider found.");
+      }
+    }
+
+    void FixedUpdate() {
+      OnPrePhysicalUpdate();
+
+      foreach (var interactionHand in _interactionHands) {
+        interactionHand.FixedUpdateHand(enableHovering, enableContact, enableGrasping);
+      }
+
+      // Apply soft contacts from both hands in unified solve.
+      // (This will clear softContacts and originalVelocities as well.)
+      if (_softContacts.Count > 0) {
+        PhysicsUtility.ApplySoftContacts(_softContacts, _softContactOriginalVelocities);
+      }
+
+      foreach (var interactionObj in _interactionBehaviours) {
+        interactionObj.FixedUpdateObject();
+      }
+
+      OnPostPhysicalUpdate();
+    }
+
+    void Update() {
+      if (Provider != null) {
+        _providerScale = Provider.transform.lossyScale.x;
+
+        foreach (var interactionHand in _interactionHands) {
+          interactionHand.HoverActivationRadius = WorldHoverActivationRadius;
+          interactionHand.TouchActivationRadius = WorldTouchActivationRadius;
+        }
+      }
+    }
+
+    void LateUpdate() {
+      OnGraphicalUpdate();
+    }
+
+    #region Object Registration
+
+
+    public void RegisterInteractionBehaviour(InteractionBehaviourBase interactionObj) {
+      _interactionBehaviours.Add(interactionObj);
+      _rigidbodyRegistry[interactionObj.Rigidbody] = interactionObj;
```

Comment:
Also looks like there is no way currently for an object to unregister itself, nor can it know if it is currently registered.